volunteer/forms.py

```diff
@@ -8,6 +11,120 @@ class VolunteerProfileForm(ModelForm):
     class Meta:
         model = VolunteerProfile
         exclude = ["user", "application_status"]
+        help_texts = {
+            "github_username": "GitHub username (e.g., username)",
+            "discord_username": "Discord username (e.g., username)",
+            "instagram_username": "Instagram username without @ (e.g., username)",
+            "bluesky_username": "Bluesky username (e.g., username or username.bsky.social)",
+            "mastodon_url": "Mastodon handle (e.g., @username@instance.tld or https://instance.tld/@username)",
+            "x_username": "X/Twitter username without @ (e.g., username)",
+            "linkedin_url": "LinkedIn URL (e.g., linkedin.com/in/username)",
+        }
+
+    def clean_github_username(self):
+        github_username = self.cleaned_data.get("github_username")
+        if github_username:
+            if not re.match(
+                r"^[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}$", github_username
+            ):
+                raise ValidationError(
+                    "GitHub username can only contain alphanumeric characters and hyphens, "
+                    "cannot start or end with a hyphen, and must be between 1-39 characters."
+                )
+        return github_username
+
+    def clean_discord_username(self):
+        discord_username = self.cleaned_data.get("discord_username")
+        if discord_username:
+            if not re.match(
+                r"^[a-zA-Z0-9](?:[a-zA-Z0-9]|[._-](?=[a-zA-Z0-9])){0,30}[a-zA-Z0-9]$",
+                discord_username,
+            ):
+                if len(discord_username) < 2 or len(discord_username) > 32:
+                    raise ValidationError(
+                        "Discord username must be between 2 and 32 characters."
+                    )
+                else:
+                    raise ValidationError(
+                        "Discord username must consist of alphanumeric characters, "
+                        "dots, underscores, or hyphens, and cannot have consecutive special characters."
+                    )
+        return discord_username
+
+    def clean_instagram_username(self):
+        instagram_username = self.cleaned_data.get("instagram_username")
+        if instagram_username:
+            if instagram_username.startswith("@"):
+                instagram_username = instagram_username[1:]
```

Comment:
I think it would be better to show an error message here, telling them that this is invalid so that they can re-enter with the correct value, instead of silently removing `@` from the username.

Reply:
This is still silently stripping the @ from instagram handle